dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracedRequestListener.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.tracing
+
+import io.opentracing.Span
+import okhttp3.Request
+
+/**
+ * Provides the ability to intercept and customise
+ * the tag values in the Span built around [Request].
+ */
+
+interface TracedRequestListener {
+
+    /**
+     * Notifies for an intercepted request and provides the [Span] that was automatically created
+     * around it. If you need to add new or override the existing tags inside this [Span] this is
+     * the right place for this operation. This function will be called before finishing and
+     * sending the [Span].
+     * @param request the intercepted [Request]
+     * @param span the [Span] created around the intercepted [Request]
+     */
+    fun onRequestIntercepted(
+        request: Request,
+        span: Span
+    ): Span
```

Comment:
Do we really want them to return a Span object ? What if they return a completely new Span ? 
I'd feel better this was a method that returned void instead.

Reply:
Same as https://github.com/DataDog/dd-sdk-android/pull/302#discussion_r445003898, so I paste it there:

> Also by expecting `return span`, we open it for eventual errors, when users do `return differentSpan` and then bad things will happen: we won't `.finish()` the original one, so propagation will not work and the timing of `differentSpan` will be almost zero milliseconds. WDYT?

Reply:
Yup you are right ....maybe not returning the span ;)